internal/controller/k8s_resources.go

```diff
@@ -96,6 +103,38 @@ func convertMetaObjectToMetadata(ctx context.Context, obj metav1.ObjectMeta, cli
 	return objMetadata, nil
 }
 
+func convertOwnerReferences(ownerReferences []metav1.OwnerReference) ([]*pb.KubernetesOwnerReference, error) {
+	if len(ownerReferences) == 0 {
+		return nil, nil
+	}
+
+	var result []*pb.KubernetesOwnerReference
```

Comment:
Pre-allocate the slice.
```suggestion
	result := make([]*pb.KubernetesOwnerReference, 0, len(ownerReferences))
```